### ISSUE TITLE
Reproducing Tulu 2.5 / Tulu 3 – PPO reward increases but IFEVAL drops (1B & 8B post-training)

### ISSUE THREAD

**Issue by RoozbehNahavandi** [allenai/open-instruct#926]
Hi all,

I’m currently trying to reproduce Tulu 2.5 and Tulu 3 models.

I’ve been using the original Tulu 2.5 repo (the version without RLVR) to run PPO training on both 1B and 8B models. While I do use the Tulu 3 dataset in some experiments, I’ve kept the training method consistent with Tulu 2.5 PPO, since my research is not focused on RLVR.

That said, I have been running into an issue: the reward steadily increases, but the IFEVAL score drops or stays flat throughout training. This happens with both:
	•	1B setup:
	•	meta-llama/Llama-3.2-1B-Instruct
	•	RM: Custom trained 1B RM
	•	Dataset: allenai/tulu-2.5-preference-data
	•	Penalty reward: -10.0
	•	Learning rate: 1e-6
	•	Episodes: 125k
	•	8B setup:
	•	allenai/Llama-3.1-Tulu-3-8B-DPO
	•	RM: allenai/Llama-3.1-Tulu-3-8B-RM
	•	Dataset: allenai/llama-3.1-tulu-3-8b-preference-mixture
	•	Penalty reward: -10.0
	•	Learning rate: 3e-7
	•	Episodes: 125k

Here are the exact configs I’m using:

To be more specific, below is the config I used:
Tulu 3 (8B) PPO Config

```
--dataset_mixer "{\"allenai/llama-3.1-tulu-3-8b-preference-mixture\": 1.0}" \
--dataset_train_splits train \
--max_token_length 1024 \
--model_name_or_path allenai/Llama-3.1-Tulu-3-8B-DPO \
--reward_model_path allenai/Llama-3.1-Tulu-3-8B-RM \
--local_rollout_forward_batch_size 1 \
--non_stop_penalty \
--stop_token eos \
--temperature 1.0 \
--chat_template tulu \
--learning_rate 3e-7 \
--total_episodes 125000 \
--penalty_reward_value -10.0 \
--per_device_train_batch_size 1 \
--per_device_eval_batch_size 1 \
--gradient_accumulation_steps 16 \
--max_prompt_token_length 512 \
--num_train_epochs 1 \
--beta 0.05
```

Tulu 2.5 (1B) PPO Config
```
--dataset_mixer "{\"allenai/tulu-2.5-preference-data\": 1.0}" \
--dataset_train_splits ultrafeedback_mean_aspects \
--dataset_eval_mixer "{\"allenai/tulu-2.5-preference-data\": 1.0}" \
--dataset_eval_splits ultrafeedback_mean_aspects \
--max_token_length 2048 \
--model_name_or_path meta-llama/Llama-3.2-1B-Instruct \
--reward_model_path path_to_rm \
--local_rollout_forward_batch_size 1 \
--non_stop_penalty \
--stop_token eos \
--temperature 1.0 \
--chat_template tulu \
--learning_rate 1e-6 \
--total_episodes 125000 \
--penalty_reward_value -10.0 \
--per_device_train_batch_size 1 \
--per_device_eval_batch_size 1 \
--gradient_accumulation_steps 4 \
--max_prompt_token_length 512 \
--num_train_epochs 1 \
--beta 0.05
```

For hardware, I’m using 4× H100s for the 1B model and 8× H100s for the 8B. I’ve tried toggling the penalty term, adjusting temperature, KL scaling, and batch sizes, but the trend remains: reward consistently goes up, while IFEVAL and a few other evaluations don’t show meaningful improvement.

A few specific questions:
	1.	Roughly how many PPO episodes were used for Tulu 2.5 / Tulu 3 training?
	2.	Have you done PPO post-training on the 1B models? If so, how did different evaluations behave?
	3.	How sensitive is PPO to the penalty_reward_value in your experience?
	4.	When reward keeps increasing but model behavior doesn’t seem to improve (e.g., IFEVAL drops or stays flat), what are the most common PPO related issues you’ve seen? Could this point to a deeper optimization problem?

If anything stands out or you have suggestions on what might be going wrong, I’d really appreciate the feedback.
Thanks again for open sourcing all of this, it’s been super helpful.

**Comment by hamishivi** [comment]
Thanks for your interest in our work!

1. For Tulu 2.5, it was 1 epoch over ultrafeedback with 1 rollout per prompt, so ~65k episodes. For Tulu 3, we trained for max 100k episodes.
2. We didn't try this explicitly. I had some early experiments with GSM8k alone for 1B that behaved well. 
3. I dont think the models are typically sensitive to `penalty_reward_value`, and in fact I think you can turn off without much effect.
4. Yeah, I've seen this with general RLHF. Actually, this matches the behaviour of IFEval I found in Unpacking DPO and PPO (https://arxiv.org/pdf/2406.09279) - look at Figure 6 and 7 for curves of how evaluations change with training. IFEval improved a tiny bit at the start, but generally was flat to worse. I think you should try instead training directly on IFEval rewards (as we did for Tulu 3) if you particularly about IFEval, or at least add ifeval-directed prompts to the prompt dataset you use during PPO.
